generators/client/templates/angular/src/main/webapp/app/layouts/profiles/_profile.service.ts

```diff
@@ -29,19 +28,23 @@ import { ProfileInfo } from './profile-info.model';
 export class ProfileService {
 
     private profileInfoUrl = <% if (authenticationType === 'uaa') { %>'<% } else { %>SERVER_API_URL + '<% } %>api/profile-info';
+    private profileInfo: Promise<ProfileInfo>;
 
     constructor(private http: Http) { }
 
-    getProfileInfo(): Observable<ProfileInfo> {
-        return this.http.get(this.profileInfoUrl)
-            .map((res: Response) => {
-                const data = res.json();
-                const pi = new ProfileInfo();
-                pi.activeProfiles = data.activeProfiles;
-                pi.ribbonEnv = data.ribbonEnv;
-                pi.inProduction = data.activeProfiles.indexOf('prod') !== -1;
-                pi.swaggerEnabled = data.activeProfiles.indexOf('swagger') !== -1;
-                return pi;
-            });
+    getProfileInfo(): Promise<ProfileInfo> {
+        if (!this.profileInfo) {
```

Comment:
I too don't understand why this wouldn't work unless the class is instantiated per call which would be strange. I dont have time to debug it and figure whats wrong but since this is not a pressing issue I guess we can wait to find a good solution

Reply:
http.get returns a new Observable, every time when it's called, so that's why currently it sends multiple requests.
If the Observable is cached, it seems, that the 'subscribe' call triggers a second request, only if, the first request is already received.
 In other places of the app, we use Promise-s, for example for the authentication, this profile info, in my opinion, should work the same

Reply:
Yes we still use promise at some places but those are mostly legacy code and there was a ticket to migrate them to observable. Any way lets wait and tackle this later